aten/src/ATen/core/Capsule.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <c10/util/intrusive_ptr.h>
+
+struct Capsule final: public c10::intrusive_ptr_target  {
```

Comment:
It was originally a `caffe2::Blob` - I had a conversation with @zdevito where the conclusion was to use a separate object. Primary reason is that Capsule has too much baggage to deal with.

More concretely, `caffe2::Blob` requires usage of `CAFFE2_KNOWN_TYPE` in several places (for the typemeta stuff), and I think that's a good reason to avoid it.